src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -56,9 +56,23 @@ protected BoosterParameter(TArgs args)
             /// </summary>
             public virtual void UpdateParameters(Dictionary<string, object> res)
             {
-                FieldInfo[] fields = Args.GetType().GetFields();
+                FieldInfo[] fields = Args.GetType().GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
                 foreach (var field in fields)
+                {
+                    var attribute = GetAttribute(field);
+
+                    if (attribute == null)
+                    {
+                        continue;
+                    }
```

Comment:
>} [](start = 20, length = 1)

stylistic nit: no { } for one line instructions

Reply:
Right, force of habit. Fixed.